openoa/plant.py

```diff
@@ -902,86 +1050,92 @@ def __attrs_post_init__(self):
-            return
-        if self.reanalysis is None:
-            self.data_validator(PlantData.reanalysis, self.reanalysis)
-            return
+    def _set_index_columns(self) -> None:
+        """Sets the index value for each of the `PlantData` objects that are not `None`."""
+        if self.scada is not None:
+            time_col = self.metadata.scada.col_map["time"]
+            id_col = self.metadata.scada.col_map["id"]
+            self.scada[time_col] = pd.DatetimeIndex(self.scada[time_col])
+            self.scada = self.scada.set_index([time_col, id_col], drop=False)
+            self.scada.index.names = ["time", "id"]
 
-        if not isinstance(self.reanalysis, dict):
-            raise ValueError(
-                "Reanalysis data should be provided as a dictionary of product name (keys) and api kwargs or data"
-            )
+        if self.meter is not None:
+            time_col = self.metadata.meter.col_map["time"]
+            self.meter[time_col] = pd.DatetimeIndex(self.meter[time_col])
+            self.meter = self.meter.set_index([time_col], drop=False)
+            self.meter.index.name = "time"
 
-        reanalysis = {}
-        for name, value in self.reanalysis.items():
-            if isinstance(value, dict):
-                value.update(
-                    dict(
-                        dataset=name,
-                        lat=self.metadata.latitude,
-                        lon=self.metadata.longitude,
-                        calc_derived_vars=True,
-                    )
-                )
-                reanalysis[name] = download_reanalysis_data_planetos(**value)
-            else:
-                reanalysis[name] = load_to_pandas(value)
+        if self.status is not None:
+            time_col = self.metadata.status.col_map["time"]
+            id_col = self.metadata.status.col_map["id"]
+            self.status[time_col] = pd.DatetimeIndex(self.status[time_col])
+            self.status = self.status.set_index([time_col, id_col], drop=False)
+            self.status.index.names = ["time", "id"]
 
-        self.reanalysis = reanalysis
-        self._calculate_reanalysis_columns()
+        if self.tower is not None:
+            time_col = self.metadata.tower.col_map["time"]
+            id_col = self.metadata.tower.col_map["id"]
+            self.tower[time_col] = pd.DatetimeIndex(self.tower[time_col])
+            self.tower = self.tower.set_index([time_col, id_col], drop=False)
+            self.tower.index.names = ["time", "id"]
+
+        if self.curtail is not None:
+            time_col = self.metadata.curtail.col_map["time"]
+            self.curtail[time_col] = pd.DatetimeIndex(self.curtail[time_col])
+            self.curtail = self.curtail.set_index([time_col], drop=False)
+            self.curtail.index.name = "time"
+
+        if self.asset is not None:
+            id_col = self.metadata.asset.col_map["id"]
+            self.asset = self.asset.set_index([id_col], drop=False)
```

Comment:
set_index with drop=false will preserve a duplicate id column

Reply:
This was more or less me being unsure if we should keep the column or not, so happy to defer to your judgement on if we should drop the column after setting the index because I could go either way with it.

Reply:
After seeing your later comment, I've removed `drop=False`.